src/internal.c

```diff
@@ -1346,7 +1346,7 @@ int wolfSSL_dtls_export_internal(WOLFSSL* ssl, byte* buf, word32 sz)
 
 
 /* On success return amount of buffer consumed */
-int wolfSSL_dtls_import_state_internal(WOLFSSL* ssl, byte* buf, word32 sz)
+int wolfSSL_dtls_import_state_internal(WOLFSSL* ssl, const byte* buf, word32 sz)
```

Comment:
scan-build is reporting the value of idx  from the line `idx += ret;` at the end of the function isn't being used. It is returning `ret` and it should probably be returning `idx`.